Added webshim for html5 date

jgreben opened the pull request:
Fixes #336 #293

Using CDN version: http://cdn.jsdelivr.net/webshim/1.15.8/polyfiller.js

jgreben commented:
Closing PR for time being. Waiting for comments from @jvine about usability and accessibility before pull request.

jkeck commented:
@jgreben it seems like you've included some upstream commits in a merge commit (62d1356fd0353a1215e90ae4afe9b00487b3ec66). I think what you may want to do is rebase your branch on top of master and fix any conflicts there.  I would be more than happy to look into this w/ you next week.

I may be completely wrong about this being a problem though, @cbeer let me know if I'm wrong here and I'll :shipit:.

jgreben commented:
I went ahead and pulled from master, rebased, and fixed a conflict in the item_selector_spec. Pushing again and making new PR. Let me know if it looks good now.

jkeck commented on app/assets/javascripts/html5_webshim.js:
```diff
@@ -0,0 +1,15 @@
+  });
+
+  window.webshim.polyfill('forms-ext');
+  window.onunload = function(){};
```
Just out of curiosity, what does this empty `window.onunload` function do for us?

jgreben replied:
The pollyfill does not work if you navigate away from the page and back again. This fixes it somehow.. I think it has something to do with the bfcache. I forgot where I came across it, maybe it was this http://stackoverflow.com/questions/158319/cross-browser-onload-event-and-the-back-button

jkeck replied:
Hmm, that seems fishy to me.  Is there any way I can get you to give me the steps to replicate that issue (or show me)?  From what I understand, binding to the `ready` and `page:load` events should handle that case.
